### ISSUE TITLE
Attempt to (de-)serialize anonymous class WARN for Jenkins Plugin

### ISSUE THREAD

**Issue by msymons** [javamelody/javamelody#768]
After upgrading Jenkins to 2.121.2 LTS and with JavaMelody plugin 1.73.1 installed,  the following is seen in Jenkins node logs

```
<===[JENKINS REMOTING CAPACITY]===>channel started
Remoting version: 3.21
This is a Unix agent
Evacuated stdout
Aug 09, 2018 11:11:50 AM org.jenkinsci.remoting.util.AnonymousClassWarnings warn
WARNING: Attempt to (de-)serialize anonymous class org.jenkinsci.plugins.envinject.EnvInjectComputerListener$2; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
Agent successfully connected and online
```

Also seen for Windows nodes connecting via JNLP.

Note link in log to [Serialization of anonymous classes](https://wiki.jenkins.io/display/JENKINS/Serialization+of+anonymous+classes) wiki page.

**Comment by evernat** [comment]
I don't think that this is related to the javamelody monitoring plugin. In fact, the class `org.jenkinsci.plugins.envinject.EnvInjectComputerListener$2` in the message is from the EnvInject plugin.
And there is already an issue created for the same thing in the EnvInject plugin:
https://issues.jenkins-ci.org/browse/JENKINS-52946

I suppose for now that it is not related to the javamelody monitoring plugin, so closing as invalid.

**Comment by msymons** [comment]
@evernat, I am afraid that I copied the wrong bit of log!  Here's a 2nd attempt,  where only the 2nd WARNING is of interest to you...

```
INFO: Connected
Aug 21, 2018 12:58:04 PM org.jenkinsci.remoting.util.AnonymousClassWarnings warn
WARNING: Attempt to (de-)serialize anonymous class hudson.FilePath$27; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
Aug 21, 2018 12:58:05 PM org.jenkinsci.remoting.util.AnonymousClassWarnings warn
WARNING: Attempt to (de-)serialize anonymous class net.bull.javamelody.RemoteCallHelper$1; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
```

**Comment by evernat** [comment]
For the first warning, I think that it does not come from the monitoring plugin and that it comes from Jenkins core and that [it was fixed](https://github.com/jenkinsci/jenkins/commit/63fa1785ebe22dcf4e62957566a5fa0587ad9ddc). So upgrading your Jenkins server should fix that.

The 2nd warning about net.bull.javamelody.RemoteCallHelper$1 is now fixed by this commit:
https://github.com/jenkinsci/monitoring-plugin/commit/a61d1f074b19245b36988c394697d243f99a8dc6
And it is ready for the next released (1.74), probably early September.
Thanks for the issue.
